webapp/src/note_rhs.tsx

```diff
@@ -0,0 +1,179 @@
+import React, {FC, useEffect, useState} from 'react';
+
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/common';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {useDispatch, useSelector} from 'react-redux';
+
+import styled from 'styled-components';
+
+import {updateUpstream, fetchNote} from 'client';
+import {makeSelectNoteValue, setNoteValue} from 'redux_connectors';
+
+// @ts-ignore
+const {formatText, messageHtmlToComponent} = window.PostUtils;
+
+const markdownOptions = {
+    singleline: false,
+    mentionHighlight: true,
+    atMentions: true,
+};
+
+const Header = styled.div`
+    display: grid;
+    grid-template-columns: 1fr auto 1fr;
+    justify-items: center;
+    font-size: 12px;
+    font-style: normal;
+    font-weight: 600;
+    line-height:47px;
+    height: 47px;
+    letter-spacing: 0;
+    text-align: center;
+    box-shadow: inset 0px -1px 0px var(--center-channel-color-24);
+`;
+
+const CenterCell = styled.div`
+    grid-column-start: 2;
+`;
+
+const EditButton = styled.span`
+    color: var(--button-color);
+    background-color: var(--button-bg);
+    padding: 8px 16px;
+    border-radius: 4px;
+    border: none;
+    cursor: pointer;
+
+    >.icon {
+        font-size: 14px;
+    }
+
+    &:hover {
+        background: linear-gradient(0deg, rgba(var(--center-channel-color-rgb), 0.16), rgba(var(--center-channel-color-rgb), 0.16)), var(--button-bg);
+        text-decoration: none;
+        color: var(--button-color);
+    }
+`;
+
+const NoteContainer = styled.div`
+    padding: 16px;
+    overflow-y: auto;
+    overflow-x: hidden;
+`;
+
+const NoteInput = styled.textarea`
+    background-color: rgb(var(--center-channel-bg-rgb));
+    border: none;
+    padding: 16px;
+    font-size: 14px;
+    resize: none;
+    flex-grow: 1;
+
+    &:focus {
+        border: none;
+        box-shadow: inset 0 0 0 2px var(--button-bg);
+    }
+    
+`;
+
+const NoNotesContainer = styled.div`
+    margin: 20px auto;
+    display: block;
+    flex-direction: column;
+    align-items: center;
+    max-width: 300px;
+
+    h1 {
```

Comment:
I am 4/5 on styling html elements, I know usually within the styled component scope it should be ok but in the case of webapp we have many conflicting styled because of different scope breaches and we don't know where this might take over specificity over another element.

Reply:
@nevyangelova 
So styled components have their own scope. This will only apply within the styled component. As far as I am aware it can't cause any conflicts outside if it's scope.
Also I think this is the only way to do this since I am not styling my own components I am styling the webapp markdown components.